x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/SecurityUsageTransportAction.java

```diff
@@ -89,78 +98,54 @@ protected void masterOperation(
         ClusterState state,
         ActionListener<XPackUsageFeatureResponse> listener
     ) {
-        Map<String, Object> sslUsage = sslUsage(settings);
-        Map<String, Object> tokenServiceUsage = tokenServiceUsage(settings);
-        Map<String, Object> apiKeyServiceUsage = apiKeyServiceUsage(settings);
-        Map<String, Object> auditUsage = auditUsage(settings);
-        Map<String, Object> ipFilterUsage = ipFilterUsage(ipFilter);
-        Map<String, Object> anonymousUsage = singletonMap("enabled", AnonymousUser.isAnonymousEnabled(settings));
-        Map<String, Object> fips140Usage = fips140Usage(settings);
-        Map<String, Object> operatorPrivilegesUsage = Map.of(
-            "available",
-            Security.OPERATOR_PRIVILEGES_FEATURE.checkWithoutTracking(licenseState),
-            "enabled",
-            OperatorPrivileges.OPERATOR_PRIVILEGES_ENABLED.get(settings)
-        );
-
-        final AtomicReference<Map<String, Object>> rolesUsageRef = new AtomicReference<>();
-        final AtomicReference<Map<String, Object>> roleMappingUsageRef = new AtomicReference<>();
-        final AtomicReference<Map<String, Object>> realmsUsageRef = new AtomicReference<>();
-
         final boolean enabled = XPackSettings.SECURITY_ENABLED.get(settings);
-        final CountDown countDown = new CountDown(3);
-        final Runnable doCountDown = () -> {
-            if (countDown.countDown()) {
-                var usage = new SecurityFeatureSetUsage(
-                    enabled,
-                    realmsUsageRef.get(),
-                    rolesUsageRef.get(),
-                    roleMappingUsageRef.get(),
-                    sslUsage,
-                    auditUsage,
-                    ipFilterUsage,
-                    anonymousUsage,
-                    tokenServiceUsage,
-                    apiKeyServiceUsage,
-                    fips140Usage,
-                    operatorPrivilegesUsage
-                );
-                listener.onResponse(new XPackUsageFeatureResponse(usage));
-            }
-        };
```

Comment:
I refactored all this `countDown` code into the `Builder` at the bottom of the file. I think it's easier to read & understand, but I'm happy to try a different approach.